File: LeetCode/2371. Minimize Maximum Value in a Grid/solution.py
from typing import List


class Solution:
    def minScore(self, grid: List[List[int]]) -> List[List[int]]:
        m, n = len(grid), len(grid[0])

        aux = sorted([(grid[i][j], i, j) for j in range(n) for i in range(m)])

        row_mins, col_mins = [0] * m, [0] * n

        for _, i, j in aux:
            grid[i][j] = max(row_mins[i], col_mins[j]) + 1

            row_mins[i] = max(row_mins[i], grid[i][j])
            col_mins[j] = max(col_mins[j], grid[i][j])

        return grid

File: LeetCode/2371. Minimize Maximum Value in a Grid/solution.swift
import Foundation

class Solution {
    func minScore(_ grid: [[Int]]) -> [[Int]] {
        let m = grid.count, n = grid[0].count
        var grid1 = Array(repeating: Array(repeating: 0, count: n), count: m)
        var rowMins = Array(repeating: 0, count: m), colMins = Array(repeating: 0, count: n)
        
        var aux:[(Int, Int, Int)] = []
        
        for i in 0..<m {
            for j in 0..<n {
                grid1[i][j] = grid[i][j]
                aux.append((grid[i][j], i, j))
            }
        }
        aux.sort { $0.0 < $1.0 }
        
        for val in aux {
            var (_, i, j) = val
            grid1[i][j] = max(rowMins[i], colMins[j]) + 1
            
            rowMins[i] = max(rowMins[i], grid1[i][j])
            colMins[j] = max(colMins[j], grid1[i][j])
        }
        return grid1
    }
}
